// @prisma/client-sqlite/wasm.js

/* !!! This is code generated by Prisma. Do not edit directly. !!!
/* eslint-disable */
// biome-ignore-all lint: generated file

Object.defineProperty(exports, "__esModule", { value: true });

const {
  PrismaClientKnownRequestError,
  PrismaClientUnknownRequestError,
  PrismaClientRustPanicError,
  PrismaClientInitializationError,
  PrismaClientValidationError,
  getPrismaClient,
  sqltag,
  empty,
  join,
  raw,
  skip,
  Decimal,
  Debug,
  objectEnumValues,
  makeStrictEnum,
  Extensions,
  warnOnce,
  defineDmmfProperty,
  Public,
  getRuntime,
  createParam,
} = require('./runtime/wasm-engine-edge.js')


const Prisma = {}

exports.Prisma = Prisma
exports.$Enums = {}

/**
 * Prisma Client JS version: 6.19.0
 * Query Engine version: 2ba551f319ab1df4bc874a89965d8b3641056773
 */
Prisma.prismaVersion = {
  client: "6.19.0",
  engine: "2ba551f319ab1df4bc874a89965d8b3641056773"
}

Prisma.PrismaClientKnownRequestError = PrismaClientKnownRequestError;
Prisma.PrismaClientUnknownRequestError = PrismaClientUnknownRequestError
Prisma.PrismaClientRustPanicError = PrismaClientRustPanicError
Prisma.PrismaClientInitializationError = PrismaClientInitializationError
Prisma.PrismaClientValidationError = PrismaClientValidationError
Prisma.Decimal = Decimal

/**
 * Re-export of sql-template-tag
 */
Prisma.sql = sqltag
Prisma.empty = empty
Prisma.join = join
Prisma.raw = raw
Prisma.validator = Public.validator

/**
* Extensions
*/
Prisma.getExtensionContext = Extensions.getExtensionContext
Prisma.defineExtension = Extensions.defineExtension

/**
 * Shorthand utilities for JSON filtering
 */
Prisma.DbNull = objectEnumValues.instances.DbNull
Prisma.JsonNull = objectEnumValues.instances.JsonNull
Prisma.AnyNull = objectEnumValues.instances.AnyNull

Prisma.NullTypes = {
  DbNull: objectEnumValues.classes.DbNull,
  JsonNull: objectEnumValues.classes.JsonNull,
  AnyNull: objectEnumValues.classes.AnyNull
}





/**
 * Enums
 */
exports.Prisma.TransactionIsolationLevel = makeStrictEnum({
  Serializable: 'Serializable'
});

exports.Prisma.AreaScalarFieldEnum = {
  id: 'id',
  nome: 'nome',
  descricao: 'descricao',
  ativo: 'ativo',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.EquipamentoScalarFieldEnum = {
  id: 'id',
  nome: 'nome',
  tag: 'tag',
  descricao: 'descricao',
  ativo: 'ativo',
  areaId: 'areaId',
  tipoId: 'tipoId',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.TipoEquipamentoScalarFieldEnum = {
  id: 'id',
  nome: 'nome',
  descricao: 'descricao',
  ativo: 'ativo',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.CheckTemplateScalarFieldEnum = {
  id: 'id',
  tipoId: 'tipoId',
  nome: 'nome',
  descricao: 'descricao',
  ordem: 'ordem',
  obrigatorio: 'obrigatorio',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.ParadaScalarFieldEnum = {
  id: 'id',
  macro: 'macro',
  nome: 'nome',
  descricao: 'descricao',
  equipeResponsavel: 'equipeResponsavel',
  dataInicio: 'dataInicio',
  dataFim: 'dataFim',
  duracaoPrevistaHoras: 'duracaoPrevistaHoras',
  status: 'status',
  tipo: 'tipo',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.TesteScalarFieldEnum = {
  id: 'id',
  paradaId: 'paradaId',
  equipamentoId: 'equipamentoId',
  status: 'status',
  observacoes: 'observacoes',
  problemaDescricao: 'problemaDescricao',
  dataTeste: 'dataTeste',
  testadoPor: 'testadoPor',
  naoAplica: 'naoAplica',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.UserScalarFieldEnum = {
  id: 'id',
  name: 'name',
  email: 'email',
  password: 'password',
  username: 'username',
  role: 'role',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  emailVerified: 'emailVerified',
  image: 'image'
};

exports.Prisma.AccountScalarFieldEnum = {
  id: 'id',
  userId: 'userId',
  type: 'type',
  provider: 'provider',
  providerAccountId: 'providerAccountId',
  expires_at: 'expires_at',
  token_type: 'token_type',
  scope: 'scope',
  refresh_token: 'refresh_token',
  access_token: 'access_token',
  id_token: 'id_token',
  session_state: 'session_state',
  oauth_token_secret: 'oauth_token_secret',
  oauth_token: 'oauth_token',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.SessionScalarFieldEnum = {
  id: 'id',
  sessionToken: 'sessionToken',
  userId: 'userId',
  expires: 'expires'
};

exports.Prisma.VerificationTokenScalarFieldEnum = {
  identifier: 'identifier',
  token: 'token',
  expires: 'expires'
};

exports.Prisma.SortOrder = {
  asc: 'asc',
  desc: 'desc'
};

exports.Prisma.NullsOrder = {
  first: 'first',
  last: 'last'
};
exports.ParadaStatus = exports.$Enums.ParadaStatus = {
  em_andamento: 'em_andamento',
  concluida: 'concluida',
  cancelada: 'cancelada'
};

exports.ParadaTipo = exports.$Enums.ParadaTipo = {
  preventiva: 'preventiva',
  corretiva: 'corretiva',
  emergencial: 'emergencial'
};

exports.TesteStatus = exports.$Enums.TesteStatus = {
  pendente: 'pendente',
  ok: 'ok',
  problema: 'problema',
  nao_aplica: 'nao_aplica'
};

exports.Prisma.ModelName = {
  Area: 'Area',
  Equipamento: 'Equipamento',
  TipoEquipamento: 'TipoEquipamento',
  CheckTemplate: 'CheckTemplate',
  Parada: 'Parada',
  Teste: 'Teste',
  User: 'User',
  Account: 'Account',
  Session: 'Session',
  VerificationToken: 'VerificationToken'
};
/**
 * Create the Client
 */
const config = {
  "generator": {
    "name": "client",
    "provider": {
      "fromEnvVar": null,
      "value": "prisma-client-js"
    },
    "output": {
      "value": "/home/marcelo/next-checklist/prisma/node_modules/@prisma/client-sqlite",
      "fromEnvVar": null
    },
    "config": {
      "engineType": "library"
    },
    "binaryTargets": [
      {
        "fromEnvVar": null,
        "value": "debian-openssl-3.0.x",
        "native": true
      }
    ],
    "previewFeatures": [],
    "sourceFilePath": "/home/marcelo/next-checklist/prisma/schema.sqlite.prisma",
    "isCustomOutput": true
  },
  "relativeEnvPaths": {
    "rootEnvPath": null,
    "schemaEnvPath": "../../../../.env"
  },
  "relativePath": "../../..",
  "clientVersion": "6.19.0",
  "engineVersion": "2ba551f319ab1df4bc874a89965d8b3641056773",
  "datasourceNames": [
    "db"
  ],
  "activeProvider": "sqlite",
  "inlineDatasources": {
    "db": {
      "url": {
        "fromEnvVar": "DATABASE_URL",
        "value": null
      }
    }
  },
  "inlineSchema": "generator client {\n  provider      = \"prisma-client-js\"\n  binaryTargets = [\"native\"]\n  output        = \"./node_modules/@prisma/client-sqlite\"\n}\n\ndatasource db {\n  provider = \"sqlite\"\n  url      = env(\"DATABASE_URL\")\n}\n\nenum ParadaStatus {\n  em_andamento\n  concluida\n  cancelada\n}\n\nenum ParadaTipo {\n  preventiva\n  corretiva\n  emergencial\n}\n\nenum TesteStatus {\n  pendente\n  ok\n  problema\n  nao_aplica\n}\n\nmodel Area {\n  id           Int           @id @default(autoincrement())\n  nome         String\n  descricao    String?\n  ativo        Boolean       @default(true)\n  equipamentos Equipamento[]\n  createdAt    DateTime      @default(now())\n  updatedAt    DateTime      @updatedAt\n}\n\nmodel Equipamento {\n  id        Int              @id @default(autoincrement())\n  nome      String\n  tag       String           @unique\n  descricao String?\n  ativo     Boolean          @default(true)\n  area      Area             @relation(fields: [areaId], references: [id])\n  areaId    Int\n  tipo      TipoEquipamento? @relation(fields: [tipoId], references: [id])\n  tipoId    Int?\n  testes    Teste[]\n  createdAt DateTime         @default(now())\n  updatedAt DateTime         @updatedAt\n}\n\nmodel TipoEquipamento {\n  id             Int             @id @default(autoincrement())\n  nome           String          @unique\n  descricao      String?\n  ativo          Boolean         @default(true)\n  equipamentos   Equipamento[]\n  checkTemplates CheckTemplate[]\n  createdAt      DateTime        @default(now())\n  updatedAt      DateTime        @updatedAt\n}\n\nmodel CheckTemplate {\n  id          Int             @id @default(autoincrement())\n  tipo        TipoEquipamento @relation(fields: [tipoId], references: [id])\n  tipoId      Int\n  nome        String\n  descricao   String?\n  ordem       Int?\n  obrigatorio Boolean         @default(true)\n  createdAt   DateTime        @default(now())\n  updatedAt   DateTime        @updatedAt\n}\n\nmodel Parada {\n  id                   Int          @id @default(autoincrement())\n  macro                String?\n  nome                 String\n  descricao            String?\n  equipeResponsavel    String?\n  dataInicio           DateTime?\n  dataFim              DateTime?\n  duracaoPrevistaHoras Int?\n  status               ParadaStatus @default(em_andamento)\n  tipo                 ParadaTipo\n  testes               Teste[]\n  createdAt            DateTime     @default(now())\n  updatedAt            DateTime     @updatedAt\n}\n\nmodel Teste {\n  id                Int         @id @default(autoincrement())\n  parada            Parada      @relation(fields: [paradaId], references: [id])\n  paradaId          Int\n  equipamento       Equipamento @relation(fields: [equipamentoId], references: [id])\n  equipamentoId     Int\n  status            TesteStatus @default(pendente)\n  observacoes       String?\n  problemaDescricao String?\n  dataTeste         DateTime?\n  testadoPor        String?\n  naoAplica         Boolean     @default(false)\n  createdAt         DateTime    @default(now())\n  updatedAt         DateTime    @updatedAt\n}\n\nmodel User {\n  id            Int       @id @default(autoincrement())\n  name          String\n  email         String    @unique\n  password      String\n  username      String?   @unique\n  role          String?\n  createdAt     DateTime  @default(now())\n  updatedAt     DateTime  @updatedAt\n  emailVerified DateTime?\n  image         String?\n  accounts      Account[]\n  sessions      Session[]\n}\n\nmodel Account {\n  id                 Int      @id @default(autoincrement())\n  userId             Int\n  type               String\n  provider           String\n  providerAccountId  String\n  expires_at         Int?\n  token_type         String?\n  scope              String?\n  refresh_token      String?\n  access_token       String?\n  id_token           String?\n  session_state      String?\n  oauth_token_secret String?\n  oauth_token        String?\n  createdAt          DateTime @default(now())\n  updatedAt          DateTime @updatedAt\n\n  user User @relation(fields: [userId], references: [id], onDelete: Cascade)\n\n  @@unique([provider, providerAccountId])\n}\n\nmodel Session {\n  id           Int      @id @default(autoincrement())\n  sessionToken String   @unique\n  userId       Int\n  expires      DateTime\n\n  user User @relation(fields: [userId], references: [id], onDelete: Cascade)\n}\n\nmodel VerificationToken {\n  identifier String\n  token      String   @unique\n  expires    DateTime\n\n  @@id([identifier, token])\n}\n",
  "inlineSchemaHash": "f9aac5736e00fb3a10f17fa8fb636f12f7af61cf28678d530d93ab04cda15f3f",
  "copyEngine": true
}
config.dirname = '/'

config.runtimeDataModel = JSON.parse("{\"models\":{\"Area\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"nome\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"descricao\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"ativo\",\"kind\":\"scalar\",\"type\":\"Boolean\"},{\"name\":\"equipamentos\",\"kind\":\"object\",\"type\":\"Equipamento\",\"relationName\":\"AreaToEquipamento\"},{\"name\":\"createdAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"updatedAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"}],\"dbName\":null},\"Equipamento\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"nome\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"tag\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"descricao\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"ativo\",\"kind\":\"scalar\",\"type\":\"Boolean\"},{\"name\":\"area\",\"kind\":\"object\",\"type\":\"Area\",\"relationName\":\"AreaToEquipamento\"},{\"name\":\"areaId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"tipo\",\"kind\":\"object\",\"type\":\"TipoEquipamento\",\"relationName\":\"EquipamentoToTipoEquipamento\"},{\"name\":\"tipoId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"testes\",\"kind\":\"object\",\"type\":\"Teste\",\"relationName\":\"EquipamentoToTeste\"},{\"name\":\"createdAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"updatedAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"}],\"dbName\":null},\"TipoEquipamento\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"nome\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"descricao\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"ativo\",\"kind\":\"scalar\",\"type\":\"Boolean\"},{\"name\":\"equipamentos\",\"kind\":\"object\",\"type\":\"Equipamento\",\"relationName\":\"EquipamentoToTipoEquipamento\"},{\"name\":\"checkTemplates\",\"kind\":\"object\",\"type\":\"CheckTemplate\",\"relationName\":\"CheckTemplateToTipoEquipamento\"},{\"name\":\"createdAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"updatedAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"}],\"dbName\":null},\"CheckTemplate\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"tipo\",\"kind\":\"object\",\"type\":\"TipoEquipamento\",\"relationName\":\"CheckTemplateToTipoEquipamento\"},{\"name\":\"tipoId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"nome\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"descricao\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"ordem\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"obrigatorio\",\"kind\":\"scalar\",\"type\":\"Boolean\"},{\"name\":\"createdAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"updatedAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"}],\"dbName\":null},\"Parada\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"macro\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"nome\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"descricao\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"equipeResponsavel\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"dataInicio\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"dataFim\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"duracaoPrevistaHoras\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"status\",\"kind\":\"enum\",\"type\":\"ParadaStatus\"},{\"name\":\"tipo\",\"kind\":\"enum\",\"type\":\"ParadaTipo\"},{\"name\":\"testes\",\"kind\":\"object\",\"type\":\"Teste\",\"relationName\":\"ParadaToTeste\"},{\"name\":\"createdAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"updatedAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"}],\"dbName\":null},\"Teste\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"parada\",\"kind\":\"object\",\"type\":\"Parada\",\"relationName\":\"ParadaToTeste\"},{\"name\":\"paradaId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"equipamento\",\"kind\":\"object\",\"type\":\"Equipamento\",\"relationName\":\"EquipamentoToTeste\"},{\"name\":\"equipamentoId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"status\",\"kind\":\"enum\",\"type\":\"TesteStatus\"},{\"name\":\"observacoes\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"problemaDescricao\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"dataTeste\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"testadoPor\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"naoAplica\",\"kind\":\"scalar\",\"type\":\"Boolean\"},{\"name\":\"createdAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"updatedAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"}],\"dbName\":null},\"User\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"name\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"email\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"password\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"username\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"role\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"createdAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"updatedAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"emailVerified\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"image\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"accounts\",\"kind\":\"object\",\"type\":\"Account\",\"relationName\":\"AccountToUser\"},{\"name\":\"sessions\",\"kind\":\"object\",\"type\":\"Session\",\"relationName\":\"SessionToUser\"}],\"dbName\":null},\"Account\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"userId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"type\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"provider\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"providerAccountId\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"expires_at\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"token_type\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"scope\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"refresh_token\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"access_token\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"id_token\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"session_state\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"oauth_token_secret\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"oauth_token\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"createdAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"updatedAt\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"user\",\"kind\":\"object\",\"type\":\"User\",\"relationName\":\"AccountToUser\"}],\"dbName\":null},\"Session\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"sessionToken\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"userId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"expires\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"user\",\"kind\":\"object\",\"type\":\"User\",\"relationName\":\"SessionToUser\"}],\"dbName\":null},\"VerificationToken\":{\"fields\":[{\"name\":\"identifier\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"token\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"expires\",\"kind\":\"scalar\",\"type\":\"DateTime\"}],\"dbName\":null}},\"enums\":{},\"types\":{}}")
defineDmmfProperty(exports.Prisma, config.runtimeDataModel)
config.engineWasm = {
  getRuntime: async () => require('./query_engine_bg.js'),
  getQueryEngineWasmModule: async () => {
    const loader = (await import('#wasm-engine-loader')).default
    const engine = (await loader).default
    return engine
  }
}
config.compilerWasm = undefined

config.injectableEdgeEnv = () => ({
  parsed: {
    DATABASE_URL: typeof globalThis !== 'undefined' && globalThis['DATABASE_URL'] || typeof process !== 'undefined' && process.env && process.env.DATABASE_URL || undefined
  }
})

if (typeof globalThis !== 'undefined' && globalThis['DEBUG'] || typeof process !== 'undefined' && process.env && process.env.DEBUG || undefined) {
  Debug.enable(typeof globalThis !== 'undefined' && globalThis['DEBUG'] || typeof process !== 'undefined' && process.env && process.env.DEBUG || undefined)
}

const PrismaClient = getPrismaClient(config)
exports.PrismaClient = PrismaClient
Object.assign(exports, Prisma)
